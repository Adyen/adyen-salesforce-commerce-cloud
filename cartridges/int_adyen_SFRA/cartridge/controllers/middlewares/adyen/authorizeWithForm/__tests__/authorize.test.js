"use strict";

jest.mock('../order');
jest.mock('../error');
jest.mock('../payment');

var OrderMgr = require('dw/order/OrderMgr');

var COHelpers = require('*/cartridge/scripts/checkout/checkoutHelpers');

var handleAuthorize = require('../authorize');

var handleOrderConfirmation = require('../order');

var handleInvalidPayment = require('../payment');

var handleError = require('../error');

var req;
var res;
beforeEach(function () {
  jest.clearAllMocks();
  req = {
    form: {
      MD: 'mocked_md'
    },
    querystring: {
      merchantReference: 'mocked_merchantReference'
    }
  };
  res = {
    redirect: jest.fn()
  };
});
afterEach(function () {
  jest.resetModules();
});
describe('Authorize', function () {
  it('should authorize when MD is valid', function () {
    OrderMgr.toArray.mockImplementation(function () {
      return [{
        custom: {
          adyenMD: 'mocked_md'
        }
      }];
    });
    handleAuthorize({
      req: req,
      res: res,
      next: jest.fn()
    });
    expect(handleOrderConfirmation.mock.calls).toMatchSnapshot();
  });
  it('should handle error when MD is invalid', function () {
    req.form.MD = 'invalid_mocked_md';
    handleAuthorize({
      req: req,
      res: res,
      next: jest.fn()
    });
    expect(handleError.mock.calls).toMatchSnapshot();
  });
  it('should handle invalid payment when result code is not Authorised', function () {
    req.form.MD = 'unauthorised_mocked_md';
    OrderMgr.toArray.mockImplementation(function () {
      return [{
        custom: {
          adyenMD: 'unauthorised_mocked_md'
        }
      }];
    });
    handleAuthorize({
      req: req,
      res: res,
      next: jest.fn()
    });
    expect(handleInvalidPayment.mock.calls).toMatchSnapshot();
  });
  it('should handle invalid payment when there is an error while placing an order', function () {
    OrderMgr.toArray.mockImplementation(function () {
      return [{
        custom: {
          adyenMD: 'mocked_md'
        }
      }];
    });
    COHelpers.placeOrder.mockImplementation(function () {
      return {
        error: true
      };
    });
    handleAuthorize({
      req: req,
      res: res,
      next: jest.fn()
    });
    expect(handleInvalidPayment.mock.calls).toMatchSnapshot();
  });
});