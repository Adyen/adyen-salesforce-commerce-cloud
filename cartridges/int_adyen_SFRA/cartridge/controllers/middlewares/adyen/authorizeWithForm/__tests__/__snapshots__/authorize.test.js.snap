// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorize should authorize when MD is valid 1`] = `
Array [
  Array [
    Object {
      "custom": Object {
        "adyenAction": null,
        "adyenMD": null,
        "adyenPaymentData": null,
      },
    },
    Object {
      "resultCode": "Authorised",
    },
    Object {
      "custom": Object {
        "Adyen_pspReference": "mocked_pspRef",
      },
      "getPaymentInstruments": [MockFunction] {
        "calls": Array [
          Array [
            "mocked_method",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Object {
              "0": Object {
                "custom": Object {
                  "adyenMD": "mocked_md",
                },
              },
              "iterator": [Function],
              "toArray": [MockFunction] {
                "calls": Array [
                  Array [],
                  Array [],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": Array [
                      Object {
                        "custom": Object {
                          "adyenMD": "mocked_md",
                        },
                      },
                    ],
                  },
                  Object {
                    "type": "return",
                    "value": Array [
                      Object {
                        "custom": Object {
                          "adyenAction": null,
                          "adyenMD": null,
                          "adyenPaymentData": null,
                        },
                      },
                    ],
                  },
                ],
              },
            },
          },
        ],
      },
      "orderNo": "mocked_orderNo",
      "orderToken": "mocked_orderToken",
      "setExportStatus": [MockFunction],
      "setPaymentStatus": [MockFunction],
    },
    Object {
      "next": [MockFunction],
      "req": Object {
        "form": Object {
          "MD": "mocked_md",
        },
        "querystring": Object {
          "merchantReference": "mocked_merchantReference",
        },
      },
      "res": Object {
        "redirect": [MockFunction],
      },
    },
  ],
]
`;

exports[`Authorize should handle error when MD is invalid 1`] = `
Array [
  Array [
    "Not a valid MD",
    Object {
      "next": [MockFunction],
      "req": Object {
        "form": Object {
          "MD": "invalid_mocked_md",
        },
        "querystring": Object {
          "merchantReference": "mocked_merchantReference",
        },
      },
      "res": Object {
        "redirect": [MockFunction],
      },
    },
  ],
]
`;

exports[`Authorize should handle invalid payment when result code is not Authorised 1`] = `
Array [
  Array [
    Object {
      "custom": Object {
        "Adyen_pspReference": "mocked_pspRef",
      },
      "getPaymentInstruments": [MockFunction] {
        "calls": Array [
          Array [
            "mocked_method",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Object {
              "0": Object {
                "custom": Object {
                  "adyenMD": "unauthorised_mocked_md",
                },
              },
              "iterator": [Function],
              "toArray": [MockFunction] {
                "calls": Array [
                  Array [],
                  Array [],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": Array [
                      Object {
                        "custom": Object {
                          "adyenMD": "unauthorised_mocked_md",
                        },
                      },
                    ],
                  },
                  Object {
                    "type": "return",
                    "value": Array [
                      Object {
                        "custom": Object {
                          "adyenAction": null,
                          "adyenMD": null,
                          "adyenPaymentData": null,
                        },
                      },
                    ],
                  },
                ],
              },
            },
          },
        ],
      },
      "orderNo": "mocked_orderNo",
      "orderToken": "mocked_orderToken",
      "setExportStatus": [MockFunction],
      "setPaymentStatus": [MockFunction],
    },
    "payment",
    Object {
      "next": [MockFunction],
      "req": Object {
        "form": Object {
          "MD": "unauthorised_mocked_md",
        },
        "querystring": Object {
          "merchantReference": "mocked_merchantReference",
        },
      },
      "res": Object {
        "redirect": [MockFunction],
      },
    },
  ],
]
`;

exports[`Authorize should handle invalid payment when there is an error while placing an order 1`] = `
Array [
  Array [
    Object {
      "custom": Object {
        "Adyen_pspReference": "mocked_pspRef",
      },
      "getPaymentInstruments": [MockFunction] {
        "calls": Array [
          Array [
            "mocked_method",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Object {
              "0": Object {
                "custom": Object {
                  "adyenMD": "mocked_md",
                },
              },
              "iterator": [Function],
              "toArray": [MockFunction] {
                "calls": Array [
                  Array [],
                  Array [],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": Array [
                      Object {
                        "custom": Object {
                          "adyenMD": "mocked_md",
                        },
                      },
                    ],
                  },
                  Object {
                    "type": "return",
                    "value": Array [
                      Object {
                        "custom": Object {
                          "adyenAction": null,
                          "adyenMD": null,
                          "adyenPaymentData": null,
                        },
                      },
                    ],
                  },
                ],
              },
            },
          },
        ],
      },
      "orderNo": "mocked_orderNo",
      "orderToken": "mocked_orderToken",
      "setExportStatus": [MockFunction],
      "setPaymentStatus": [MockFunction],
    },
    "placeOrder",
    Object {
      "next": [MockFunction],
      "req": Object {
        "form": Object {
          "MD": "mocked_md",
        },
        "querystring": Object {
          "merchantReference": "mocked_merchantReference",
        },
      },
      "res": Object {
        "redirect": [MockFunction],
      },
    },
  ],
]
`;
